.github/workflows/test.yml

```diff
@@ -3,45 +3,84 @@ name: Run DTS tests
 on:
   pull_request:
     branches:
-      - 'main'
+      - "main"
 jobs:
   build-dts:
     uses: ./.github/workflows/build.yml
     with:
       cacheless: false
+
   run-tests:
     name: Run DTS tests
     needs: build-dts
     runs-on:
       labels: dts-builder
+    strategy:
+      matrix:
+        include:
+          - test: msi_comm
+            dpp_download_key: $MSI_DOWNLOAD
+            dpp_logs_key: $MSI_LOGS
+            log_file: output_msi.log
+
+          - test: msi_dpp
+            dpp_download_key: $MSI_DOWNLOAD
+            dpp_logs_key: $MSI_LOGS
+            log_file: output_msi.log
+
+          - test: msi_heads
+            dpp_download_key: $MSI_HEADS_DOWNLOAD
+            dpp_logs_key: $MSI_HEADS_LOGS
+            log_file: output_msi_heads.log
+
+          - test: optiplex_dpp
+            dpp_download_key: $OPTIPLEX_DOWNLOAD
+            dpp_logs_key: $OPTIPLEX_LOGS
+            log_file: output_nc_heads.log
+
+          - test: novacustom_heads
+            dpp_download_key: $OPTIPLEX_DOWNLOAD
+            dpp_logs_key: $OPTIPLEX_LOGS
+            log_file: output_nc_heads.log
+
+          - test: novacustom_comm
+            dpp_download_key: $OPTIPLEX_DOWNLOAD
+            dpp_logs_key: $OPTIPLEX_LOGS
+            log_file: output_nc_heads.log
+
+          - test: pcengines_dpp
+            dpp_download_key: $PCENGINES_DOWNLOAD
+            dpp_logs_key: $PCENGINES_LOGS
+            log_file: output_pcengines.log
+
+          - test: pcengines_seabios
+            dpp_download_key: $PCENGINES_SEABIOS_DOWNLOAD
+            dpp_logs_key: $PCENGINES_SEABIOS_LOGS
+            log_file: output_pcengines_seabios.log
+
     steps:
       - name: Checkout OSFV repo
         uses: actions/checkout@v4
         with:
-          repository: 'Dasharo/open-source-firmware-validation'
-          path: 'open-source-firmware-validation'
-          submodules: 'recursive'
-          ref: 'develop'
+          repository: "Dasharo/open-source-firmware-validation"
+          path: "open-source-firmware-validation"
+          submodules: "recursive"
+          ref: "develop"
+
       - name: Run IPXE server
         shell: bash
-        run: |
-          mkdir ipxe
-          cp build/tmp/deploy/images/genericx86-64/dts-base-image-genericx86-64.cpio.gz ipxe
-          cp build/tmp/deploy/images/genericx86-64/bzImage ipxe
-          echo -e "\n
-          #!ipxe\n
-          imgfetch --name file_kernel bzImage\n
-          imgfetch --name file_initrd dts-base-image-genericx86-64.cpio.gz\n
-          kernel file_kernel root=/dev/nfs initrd=file_initrd\n
-          boot" > ipxe/dts.ipxe
-          cd ipxe && python3 -m http.server 4321 &
+        run: ./open-source-firmware-validation/scripts/ci/ipxe-run.sh \
+          build/tmp/deploy/images/genericx86-64/dts-base-image-genericx86-64.cpio.gz \
+          build/tmp/deploy/images/genericx86-64/bzImage
+
       - name: Install requirements
```

Comment:
You can do this only once as we are using the same folder between jobs (unfortunately). I'm not sure if it will work if `venv` already exists

Reply:
I figured out a rather simple solution: 447c744eda1afb6755bff1aaccfeda07d4dfdb03

Also tested it locally with act